tests/pdb_test.go

```diff
@@ -0,0 +1,115 @@
+package tests_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	policyv1 "k8s.io/api/policy/v1"
+	"k8s.io/utils/pointer"
+
+	expect "github.com/google/goexpect"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	kvpointer "kubevirt.io/kubevirt/pkg/pointer"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/exec"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libpod"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[sig-compute]Pod Disruption Budget (PDB)", decorators.SigCompute, func() {
+	var (
+		vmi *v1.VirtualMachineInstance
+		err error
+		pdb *policyv1.PodDisruptionBudget
+	)
+
+	BeforeEach(func() {
+		vmi = libvmifact.NewCirros(libvmi.WithLogSerialConsole(true))
+		vmi.Spec.Domain.Devices.AutoattachSerialConsole = pointer.Bool(true)
+		vmi.ObjectMeta.Labels = map[string]string{}
+		vmi.ObjectMeta.Labels["kubevirt.io/vmi"] = vmi.Name
+		vmi.Spec.EvictionStrategy = kvpointer.P(v1.EvictionStrategyLiveMigrate)
+
+		By("Creating a VMI")
+		vmi, err = kubevirt.Client().VirtualMachineInstance(testsuite.NamespaceTestDefault).Create(context.Background(), vmi, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Waiting for VMI to be created")
+		Eventually(func() (*v1.VirtualMachineInstance, error) {
+			return kubevirt.Client().VirtualMachineInstance(testsuite.NamespaceTestDefault).Get(context.Background(), vmi.Name, metav1.GetOptions{})
+		}, 60*time.Second, 1*time.Second).Should(BeAssignableToTypeOf(&v1.VirtualMachineInstance{}))
+
+		By("Creating a PodDisruptionBudget for the VMI")
+		pdb = &policyv1.PodDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pdb-for-vmi",
+				Namespace: testsuite.NamespaceTestDefault,
+				OwnerReferences: []metav1.OwnerReference{
+					*metav1.NewControllerRef(vmi, v1.VirtualMachineInstanceGroupVersionKind),
+				},
+			},
+			Spec: policyv1.PodDisruptionBudgetSpec{
+				MinAvailable: &intstr.IntOrString{IntVal: 1},
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{"kubevirt.io/vmi": vmi.Name},
+				},
+			},
+		}
+		_, err = kubevirt.Client().PolicyV1().PodDisruptionBudgets(testsuite.NamespaceTestDefault).Create(context.Background(), pdb, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
```

Comment:
Why do you need this AfterEach()? Log collection is already taken care of by the test-suite.

Reply:
removed it